controllers/fenceagentsremediation_controller.go

```diff
@@ -168,10 +170,21 @@ func buildFenceAgentParams(far *v1alpha1.FenceAgentsRemediation) ([]string, erro
 		return nil, errors.New(errorMissingParams)
 	}
 	var fenceAgentParams []string
+	logger := ctrl.Log.WithName("build-fa-parameters")
+	// add shared parameters except the action parameter
 	for paramName, paramVal := range far.Spec.SharedParameters {
-		fenceAgentParams = appendParamToSlice(fenceAgentParams, paramName, paramVal)
+		if paramName == parameterActionName {
+			if paramVal != parameterActionValue {
+				logger.Info("FAR doesn't support any other action than reboot", "action", paramVal)
+			}
+		} else {
+			fenceAgentParams = appendParamToSlice(fenceAgentParams, paramName, paramVal)
+		}
 	}
+	// ensure the FA uses the reboot action
+	fenceAgentParams = appendParamToSlice(fenceAgentParams, parameterActionName, parameterActionValue)
```

Comment:
- is the action name for triggering a reboot the same for all fence agents?
- is reboot the only action which makes sense for us? What about shutdown / switch off? That would ensure fencing, just without the chance to recover. Valid usecase IMHO, when users don't want to rejoin unhealthy nodes but investigate it manually instead 🤔 

> By doing this we make sure a reboot will occur regardless of the user input

agree, that's not a good idea

> I can think of using status maybe as a method to check the node's status after the FA execution return success

I do not understand? 🤔 

> so we can be sure that the node status is back to Ready.

That might never happen, the node can have a permanent issue which isn't fixed by a reboot

Reply:
>is the action name for triggering a reboot the same for all fence agents?

Yes, every FA which uses the [fence_action function](https://github.com/ClusterLabs/fence-agents/blob/main/lib/fencing.py.py#L939-L1064) so all the FA have **reboot** as the action name to trigger **off** and then **on** for a node.

Reply:
the logic needed to exclude any action except reboot is a bit unclear to me. Just as idea, would it be possible to validate the spec with a webhook instead? I guess this way it would be clear since the beginning what's accepted and what's not.

Reply:
> What about shutdown / switch off? That would ensure fencing, just without the chance to recover. Valid usecase IMHO, when users don't want to rejoin unhealthy nodes but investigate it manually instead

Good point. I prefer to only enable `reboot` for now, and for next release support the `off` action which is indeed a valid use case.

Reply:
> would it be possible to validate the spec with a webhook instead?

I am unsure if it is possible, since the action is not an independent field in spec, and it is part of the [SharedParameters](https://github.com/medik8s/fence-agents-remediation/blob/main/api/v1alpha1/fenceagentsremediation_types.go#L42-L44) . But it does worth thinking to maybe separate it, and then use webhook

Reply:
I think you could check it if and only if it is set